buildlib/tools/common.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash -eExl
+
+WORKSPACE=${WORKSPACE:=$PWD}
+ucx_inst=${WORKSPACE}/install
+COV_MODULE="tools/cov"
+CUDA_MODULE="dev/cuda-latest"
+GDRCOPY_MODULE="dev/gdrcopy-latest"
+JDK_MODULE="dev/jdk"
+MVN_MODULE="dev/mvn"
+XPMEM_MODULE="dev/xpmem-latest"
+
+#
+# Parallel build command runs with 4 tasks, or number of cores on the system,
+# whichever is lowest
+#
+num_cpus=$(lscpu -p | grep -v '^#' | wc -l)
+[ -z $num_cpus ] && num_cpus=1
+parallel_jobs=4
+[ $parallel_jobs -gt $num_cpus ] && parallel_jobs=$num_cpus
+num_pinned_threads=$(nproc)
+[ $parallel_jobs -gt $num_pinned_threads ] && parallel_jobs=$num_pinned_threads
+
+MAKE="make"
+MAKEP="make -j${parallel_jobs}"
+export AUTOMAKE_JOBS=$parallel_jobs
+
+#
+# Set initial port number for client/server applications
+#
+server_port=$((10000 + (1000 * EXECUTOR_NUMBER)))
+
+#
+# Override maven repository path, to cache the downloaded packages accross tests
+#
+export maven_repo=${WORKSPACE}/.deps
+
+#
+# Set up parallel test execution - "worker" and "nworkers" should be set by jenkins
+#
+if [ -z "$worker" ] || [ -z "$nworkers" ]
+then
+	worker=0
+	nworkers=1
+fi
+echo "==== Running on $(hostname), worker $worker / $nworkers ===="
+
+#
+# cleanup ucx
+#
+make_clean() {
+	rm -rf ${ucx_inst}
+	$MAKEP ${1:-clean}
+}
+
+#
+# Test if an environment module exists and load it if yes.
+# Otherwise, return error code.
+#
+module_load() {
+	set +x
+	module=$1
+
+	if module avail -t 2>&1 | grep -q "^$module\$"
+	then
+		module load $module
+		set -x
+		return 0
+	else
+		set -x
+		return 1
+	fi
+}
+
+#
+# Safe unload for env modules (even if it doesn't exist)
+#
+module_unload() {
+	module=$1
+	module unload "${module}" || true
+}
+
+modules_for_coverity() {
+	res=0
+	module_load tools/cov
+	res=$(($res+$?))
+	module_load dev/cuda-latest
+	res=$(($res+$?))
+	module_load dev/gdrcopy-latest
+	res=$(($res+$?))
+    module_load dev/jdk
+    res=$(($res+$?))
+    module_load dev/mvn
+    res=$(($res+$?))
+    module_load dev/xpmem-latest
+    res=$(($res+$?))
+    return $res
```

Comment:
1. indent
2. should be in coverity script (not common)

Reply:
ok, and will use variables for module name